runtime/src/snapshot_package.rs

```diff
@@ -16,14 +13,70 @@ use solana_sdk::hash::Hash;
 use std::{
     fs,
     path::{Path, PathBuf},
-    sync::mpsc::{Receiver, SendError, Sender},
+    sync::{
+        mpsc::{Receiver, SendError, Sender},
+        Arc, Mutex,
+    },
 };
 use tempfile::TempDir;
 
+/// The sender side of the AccountsPackage channel, used by AccountsBackgroundService
 pub type AccountsPackageSender = Sender<AccountsPackagePre>;
+
+/// The receiver side of the AccountsPackage channel, used by AccountsHashVerifier
 pub type AccountsPackageReceiver = Receiver<AccountsPackagePre>;
+
+/// The error type when sending an AccountsPackage over the channel fails
 pub type AccountsPackageSendError = SendError<AccountsPackagePre>;
 
+/// The PendingSnapshotPackage passes a SnapshotPackage from AccountsHashVerifier to SnapshotPackagerService for archiving
+pub type PendingSnapshotPackage = Arc<Mutex<Option<SnapshotPackage>>>;
+
+/// This enum wraps an AccountsPackage to differentiate between a full snapshot package and an
+/// incremental snapshot package
+pub enum SnapshotPackage {
+    FullSnapshotPackage(AccountsPackage),
+    IncrementalSnapshotPackage(AccountsPackage),
+}
```

Comment:
Because the struct fields are basically the same here, and we just want to distinguish between `FullSnapshotPackage` and `IncrementalSnapshotPackage`, it might make more sense to use the following pattern

```
struct AccountsPackage<T> {
     <rest of the fields>
     _phantom: PhantomData<T>,
}
```

Here's a similar example: https://github.com/solana-labs/solana/blob/v1.7/core/src/cluster_nodes.rs#L26-L36

This makes common functions like `fn snapshot_archive_info(&self)` easier to implement without needing to case on every possible enum option.

Reply:
I agree! I've been working on the next part, with AccountsHashVerifier, and I've basically done that, but as an enum field instead of the Phantom so that I can limit (plus easily match on) the variants:

https://github.com/solana-labs/solana/blob/129dab9cdf74f43931c87c2d2a0f36faeb64e033/runtime/src/snapshot_package.rs#L188-L189

(Here's all the changes: https://github.com/solana-labs/solana/pull/19215) _<--- that_

What do you think about _that_ version? I can pull that into this PR if you like it.

Reply:
The advantage of the Phantom type is it's zero length and you can have separate `impl` blocks for the different variants, which introduces more clarity.

Reply:
Cool! Will do.